frontend_tests/node_tests/integration_bot_widget.js

```diff
@@ -0,0 +1,111 @@
+set_global('page_params', {
+    is_admin: false,
+    realm_users: [],
+});
+
+set_global('$', global.make_zjquery());
+
+set_global('channel', {
+});
+
+add_dependencies({
+    people: 'js/people.js',
+    stream_data: 'js/stream_data.js',
+});
+
+set_global('blueslip', {});
+
+var integration_bot_widget = require('js/integration_bot_widget.js');
+
+(function test_set_integration_bot_url() {
+    var success_callback;
+    var error_callback;
+    var posted;
+
+    var form_data = {
+        append: function (field, val) {
+            form_data[field] = val;
+        },
+    };
+
+    set_global('csrf_token', 'token-stub');
+    set_global('jQuery', {
+        each: function (lst, f) {
+            _.each(lst, function (v, k) {
+                f(k, v);
+            });
+        },
+    });
+
+    set_global('FormData', function () {
+        return form_data;
+    });
+
+    channel.post = function (req) {
+        posted = true;
+        assert.equal(req.url, '/json/bots');
+        success_callback = req.success;
+        error_callback = req.error;
+    };
+
+    var external_api_uri_subdomain = "https://chat.zulip.org/api/v1/external/";
+    var integration_url = "airbrake";
+    var bot_full_name = "airbrake";
+    var bot_short_name = "airbrake";
+    var stream_name = "Airbrake";
+
+    var airbrake = {
+        subscribed: false,
+        color: 'blue',
+        name: 'Airbrake',
+        stream_id: 1,
+        in_home_view: false,
+    };
+
+    // Add stream to sub
+    stream_data.add_sub(stream_name, airbrake);
```

Comment:
It might make sense to move this stream addition and its related object declaration(`var airbrake`) to reside outside this function in the global scope, if you feel that the original code file `integration_bot_widget.js` is going to expand to have more function and tests for which would require a similar stream to be added.
(I know this addition of stream would still be available to those tests which lie after this one and hence get executed later but stylistically it might be better to have this stream addition done outside if we are going to be using it again)